### ISSUE TITLE
Literal type in a type arg of an imported variable are forgotten inside function bodies

### ISSUE THREAD

**Issue by fergusq** [microsoft/pyright#1106]
**Describe the bug**
I have a list defined in file `a.py` with explicit type signature `List[Literal[1, 2]]`. After importing this variable in a different file `b.py`, its type is correctly `List[Literal[1, 2]]` as expected. However, inside functions its type is suddenly `List[int]`.

**To Reproduce**
Create files `a.py` and `b.py` as below and run `pyright b.py`

**Expected behavior**
I would expect that the type of the variable is the same both outside and inside functions.

**Screenshots or Code**
a.py:
```python
from typing import List, Literal

d: List[Literal[1, 2]] = [1]
```

b.py:
```python
from .a import d

z: Literal[1, 2]
z = d[0]

def f():
    x: Literal[1, 2]
    x = d[0]
```

The error:
```
  8:9 - error: Expression of type "int" cannot be assigned to declared type "Literal[1, 2]"
    Type "int" cannot be assigned to type "Literal[1, 2]"
      "int" cannot be assigned to type "Literal[1]"
      "int" cannot be assigned to type "Literal[2]" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
I use pyright on command-line:
```
$ pyright --version       
pyright 1.1.79
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.80, which I just published. It will also be in the next release of pylance.
